core/src/test/java/com/backblaze/b2/json/B2JsonTest.java

```diff
@@ -2792,6 +2793,104 @@ public void testClassWithLotsOfFields() throws IOException, B2JsonException {
         assertEquals("b2", obj3.field77);
     }
 
+    /**
+     * Cannot directly test the case with MAX_ARRAY_SIZE being (Integer.MAX_VALUE - 8)
+     * since the real OutOfMemory error would occur without triggering the intended
+     * IOException ("Requested array size exceeds maximum limit"). This happens even
+     * when heap size is increased to 8GB. This happens probably because the JVM heap
+     * is probably heavily allocated before the OutputStream array size is anywhere
+     * close to half of MAX_ARRAY_SIZE.
+     *
+     * Instead, create a subclass of B2JsonByteArrayOutputStream with lowered
+     * MAX_ARRAY_SIZE (1000) for testing purpose: IOException will then be thrown
+     */
+    private static class B2JsonByteArrayOutputStreamForTest extends B2JsonByteArrayOutputStream {
+        private static final int MAX_ARRAY_SIZE = 1000;
+
+        @Override
+        protected void grow(int minCapacity) throws IOException {
+            // overflow-conscious code
+            int oldCapacity = buf.length;
+            int newCapacity = oldCapacity << 1;
+            if (newCapacity - minCapacity < 0)
+                newCapacity = minCapacity;
+            if (newCapacity - MAX_ARRAY_SIZE > 0) {
+                throw new IOException("Requested array size exceeds maximum limit");
+            }
+            buf = Arrays.copyOf(buf, newCapacity);
+        }
+    }
+
+    /* convenience object for testing IOException("Requested array size exceeds maximum limit") */
+    private static class ObjectWithSomeName {
+       @B2Json.required
+       private final String name;
+
+       @B2Json.constructor(params = "name")
+       public ObjectWithSomeName(String name) {
+           this.name = name;
+       }
+
+       @Override
+       public boolean equals(Object o) {
+           if (this == o) {
+               return true;
+           }
+           if (o == null || getClass() != o.getClass()) {
+               return false;
+           }
+           final ObjectWithSomeName objectWithSomeName = (ObjectWithSomeName) o;
+           return this.name.equals(objectWithSomeName.name);
+       }
+
+       @Override
+       public int hashCode() {
+           return name.hashCode();
+       }
+    }
+
+    @Test
+    public void testObjectWithNameToJson() throws B2JsonException, IOException {
+        // an object not big enough to cause IOException("Requested array size exceeds maximum limit")
```

Comment:
how does this differ from the other tests in this file where we're not overflowing the limit?

Reply:
thinking about it more: we can get rid of this part.